### PR TITLE
Cascade role-session-name to sub-workflows

### PR DESCRIPTION
**Pull request by brucehoff**
The error that occurred in the root workflow, `deploy.yaml` also occurs in the sub-workflows.  This change propagates the fix (managing the role session name length) to the sub-workflows.

### CODE REVIEW

File: .github/workflows/aws-deploy.yaml [hunk 1]
```diff
@@ -10,6 +10,9 @@ on:
       role-to-assume:
         required: true
         type: string
+      role-session-name:
```
Comment (zaro0508) [inline .github/workflows/aws-deploy.yaml:13]: this PR is not complete.  you need to actually pass the role session name in from the deploy.yaml template.

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: .github/workflows/aws-deploy.yaml (hunk 1 above)
Comment (brucehoff) [reply .github/workflows/aws-deploy.yaml:13]: @zaro0508 I don't understand this comment.  Isn't the change to `deploy.yaml`, below, the required change?

Comment (zaro0508) [reply .github/workflows/aws-deploy.yaml:13]: yes, you are correct.  i somehow missed it.  sorry.

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -70,117 +70,136 @@ jobs:
     uses: "./.github/workflows/aws-deploy.yaml"
     with:
       role-to-assume: "arn:aws:iam::531805629419:role/sagebase-github-oidc-sage-bionetworks-it"
+      role-session-name: ${{ steps.role-session-name.outputs.substring }}
```
Comment (zaro0508) [inline .github/workflows/deploy.yaml:73]: can you reference a `step` from a different job?  i think `step` here is not in the same scope as the one that's in `org-formation` job.  you probably need to add that `Get role session name` to every one of these jobs?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yaml (hunk 2 above)
Comment (brucehoff) [reply .github/workflows/deploy.yaml:73]: Yes, you are right.  I have to find a way to pass the value between jobs.

Comment (brucehoff) [reply .github/workflows/deploy.yaml]: Done!

**Review by ConsoleCatzirl — APPROVED** [review]

**Pull request merged**